### PR TITLE
[SYCL][ESIMD] Move BFN function from experimental namespace

### PR DESCRIPTION
**Pull request by turinevgeny**

### CODE REVIEW

**Comment by turinevgeny** [comment]
I have not found a good way to not replicate the operators for `bfn_t` enum in non-experimental namespace, if you have an idea, please let me know.

**Comment by v-klochkov** [comment]
> I have not found a good way to not replicate the operators for `bfn_t` enum in non-experimental namespace, if you have an idea, please let me know.

Can you do it the same way as it was done here?
{code}
using argument_type
    __SYCL_DEPRECATED("use sycl::ext::intel::esimd::xmx::dpas_argument_type") =
        __ESIMD_NS::xmx::dpas_argument_type;
{code}

See https://github.com/intel/llvm/commit/91e8364164c6ef62a90ba62b3e11405d132270c4#diff-8040cee0a7937fbdfbdb85880bdf2873b7405fad4b20a0edff46a94dd4c6219d

**Comment by turinevgeny** [comment]
> > I have not found a good way to not replicate the operators for `bfn_t` enum in non-experimental namespace, if you have an idea, please let me know.
> 
> Can you do it the same way as it was done here? {code} using argument_type __SYCL_DEPRECATED("use sycl::ext::intel::esimd::xmx::dpas_argument_type") = __ESIMD_NS::xmx::dpas_argument_type; {code}
> 
> See [91e8364#diff-8040cee0a7937fbdfbdb85880bdf2873b7405fad4b20a0edff46a94dd4c6219d](https://github.com/intel/llvm/commit/91e8364164c6ef62a90ba62b3e11405d132270c4#diff-8040cee0a7937fbdfbdb85880bdf2873b7405fad4b20a0edff46a94dd4c6219d)

That works, thanks!

File: sycl/include/sycl/ext/intel/experimental/esimd/math.hpp [hunk 1]
```diff
@@ -1724,38 +1694,13 @@ static constexpr bfn_t operator^(bfn_t x, bfn_t y) {
...
-    // Odd number of 1-byte elements.
-    __ESIMD_NS::simd<T, N + 1> Src0, Src1, Src2;
-    Src0.template select<N, 1>() = src0;
-    Src1.template select<N, 1>() = src1;
-    Src2.template select<N, 1>() = src2;
-    auto Result = __ESIMD_ENS::bfn<FuncControl>(Src0, Src1, Src2);
-    return Result.template select<N, 1>();
-  }
+  return __ESIMD_NS::bfn<static_cast<__ESIMD_NS::bfn_t>(FuncControl)>(
+      src0, src1, src2);
```
Comment (v-klochkov) [inline sycl/include/sycl/ext/intel/experimental/esimd/math.hpp:1703]: This static_cast is not needed after added that 'using' statement.
```suggestion
  return __ESIMD_NS::bfn<FuncControl>(src0, src1, src2);
```

Comment (turinevgeny) [reply sycl/include/sycl/ext/intel/experimental/esimd/math.hpp]: Right, thanks.

**Review by v-klochkov — APPROVED** [review]

**Comment by v-klochkov** [comment]
@turinevgeny please resolve the conflicts with head.

**Comment by turinevgeny** [comment]
> @turinevgeny please resolve the conflicts with head.

Done.

**Pull request merged**